### PR TITLE
Add supported providers annotation

### PR DESCRIPTION
**Pull request by SUPERCILEX**
@samtstern While doing some debugging, I made a typo and passed in a hardcoded string to the `IdpBuilder#setProvider` method that ended in a runtime exception. 😢 I think we can be fancier with our supported providers so this PR add our own `SupportedProvider` annotation. 😎:smile:

Here's an example error:
image

PS: sorry for the yucky commit history, I built it off another PR.

### CODE REVIEW

File: auth/src/main/java/com/firebase/ui/auth/AuthUI.java [hunk 1]
```diff
@@ -73,6 +74,13 @@
  * for examples on how to get started with FirebaseUI Auth.
  */
 public class AuthUI {
+    @StringDef({
+                       EmailAuthProvider.PROVIDER_ID, EMAIL_PROVIDER,
+                       GoogleAuthProvider.PROVIDER_ID, GOOGLE_PROVIDER,
+                       FacebookAuthProvider.PROVIDER_ID, FACEBOOK_PROVIDER,
+                       TwitterAuthProvider.PROVIDER_ID, TWITTER_PROVIDER
+               })
+    public @interface SupportedProvider {}
```
Comment (SUPERCILEX) [inline auth/src/main/java/com/firebase/ui/auth/AuthUI.java]: Do we want this to be public? I feel like it could be useful, but it also means we have to maintain backwards compatibility...

File: auth/src/main/java/com/firebase/ui/auth/util/signincontainer/SmartLockBase.java [hunk 2]
```diff
@@ -132,10 +134,6 @@ public void onStop() {
     @Override
     public void onDestroy() {
         super.onDestroy();
-        cleanup();
-    }
-
-    public void cleanup() {
```
Comment (SUPERCILEX) [inline auth/src/main/java/com/firebase/ui/auth/util/signincontainer/SmartLockBase.java]: @samtstern What size level of changes do you want me to split into other PRs? I'm happy to move this to another PR. (it was just a small nit Intellij complained about)

**Comment by SUPERCILEX** [comment]
@samtstern do you want to take a look at this one before 2.0? I think it's a nice improvement to help devs know what they should be passing in for the provider.

**Comment by samtstern** [comment]
@SUPERCILEX yep I'll add this one to 2.0 milestone

**Comment by SUPERCILEX** [comment]
@samtstern cool!

**Comment by samtstern** [comment]
Forgot about this one, gonna merge it!

**Comment by SUPERCILEX** [comment]
Awesome possum! Gonna go merge stuff now 😃
